Kudu.Core/SiteExtensions/InstallationLock.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Globalization;
+using System.IO;
+using System.Threading.Tasks;
+using Kudu.Core.Infrastructure;
+
+namespace Kudu.Core.SiteExtensions
+{
+    public class InstallationLock : LockFile
+    {
+        private InstallationLock(string path)
+            : base(path)
+        { }
+
+        /// <summary>
+        /// Hold on the the lock till action complete (success or fail)
+        /// </summary>
+        public async Task<T> Hold<T>(Func<Task<T>> f)
+        {
+            try
+            {
+                await base.LockAsync();
```

Comment:
See if we can reuse https://github.com/projectkudu/kudu/blob/master/Kudu.Contracts/Infrastructure/LockExtensions.cs